pkg/operator/vspherecontroller/driver_starter.go

```diff
@@ -35,9 +36,9 @@ func (c *VSphereController) createCSIDriver() {
 		c.eventRecorder,
 	).WithLogLevelController().WithManagementStateController(
 		driverOperandName,
-		false,
-	).WithStaticResourcesController(
-		"VMwareVSphereDriverStaticResourcesController",
+		true,
+	).WithConditionalStaticResourcesController(
+		"VMwareVSphereDriverConditionalStaticResourcesController",
```

Comment:
We've seen some problems after renaming a controller name before: https://issues.redhat.com//browse/OCPBUGS-34590

The problem was that controller name changed and it's conditions were not getting cleared, then a degraded condition was kept which eventually caused upgrade failure while the cluster was not actually degraded. But this patch changes the name *and* the controller itself so it might behave different, are we sure the we won't hit the same issue and unintentionally keep conditions forever with `"VMwareVSphereDriverStaticResourcesController"` name in this case?